corehq/apps/data_cleaning/filters.py

```diff
@@ -96,6 +104,55 @@ def get_value_for_db(self):
         value = self.get_value(self.request, self.domain)
         return None if value == self._get_default_db_value() else value
 
+    @classmethod
+    def filter_query(cls, query, pinned_filter):
+        """
+        todo: it would be nice to de-duplicate the logic here
+        with the logic in CaseListReport, but we can address that at a later date.
+        """
+        couch_user = CouchUser.get_by_username(pinned_filter.session.user.username)
+        domain = pinned_filter.session.domain
+        can_access_all_locations = couch_user.has_permission(
+            domain, 'access_all_locations'
+        )
+        emwf_slugs = pinned_filter.value or cls._get_default_db_value()
+
+        if can_access_all_locations and cls.show_all_data(emwf_slugs):
+            # don't apply any case owner filters
+            return query
+
+        case_owner_filters = []
+
+        if can_access_all_locations and cls.show_project_data(emwf_slugs):
```

Comment:
This code makes me think the "Project Data" filter does nothing for location-restricted users. Should there be a condition for `not can_access_all_locations`? Or do location-restricted users not see that filter in the UI?

Same question for showing deactivated data.

Reply:
I see `test_deactivated_location_restricted` in the next commit, but it expects `query_location_restricted_cases` and it looks like that will only return active users, based on [this line](https://github.com/dimagi/commcare-hq/blob/5f1bead10d8d5d44c20a718094a623dc6314aa0b/corehq/apps/reports/standard/cases/utils.py#L172) using `active_objects`.

Reply:
@orangejenny I'm just following the same structure for the `CaseListMixin` `_build_query` logic
https://github.com/dimagi/commcare-hq/blob/79f77cce9b66acbe9e75a08f533450428b2baac3/corehq/apps/reports/standard/cases/basic.py#L74-L78

If we want to do something else, I think that can be addressed separately and in both places. I just want the functionality to be mirrored. Also, if project data isn't available in the options, it could still be a possibility for someone to try and inject that option in the filter value if they have location restriction enabled. So this line is definitely needed to address that potential security issue.

Reply:
I see, thanks. That behavior seems counterintuitive, but I agree with not wanting to address it here.